### PR TITLE
MBS-9865 Add pretty plain text build verdict

### PR DESCRIPTION
**Pull request by sergioboi**

### CODE REVIEW

File: subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/TaskErrorOutputCaptureExecutionListener.kt [hunk 1]
```diff
@@ -14,13 +14,14 @@ import org.jetbrains.kotlin.gradle.internal.KaptTask
...
-    private val logs: MutableMap<Path, StringBuilder>
+    private val logs: MutableMap<Path, StringBuilder>,
+    private val logger: CILogger
 ) : TaskExecutionListener, BuildAdapter() {
 
     override fun beforeExecute(task: Task) {
         when (task) {
             is Test -> {
-                task.ciLogger.debug("Add error listener to the task ${task.path}")
+                logger.debug("Add error listener to the task ${task.path}")
```
Comment (sergioboi) [inline subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/TaskErrorOutputCaptureExecutionListener.kt:24]: fix unrelated bug - ci folder in the Teamcity build artifacts contains a lot of task-log files

File: subprojects/gradle/build-verdict/src/test/kotlin/com/avito/android/build_verdict/BuildVerdictPluginTest.kt [hunk 2]
```diff
@@ -12,13 +11,14 @@ import com.google.gson.GsonBuilder
 import org.junit.jupiter.api.Test
 import org.junit.jupiter.api.io.TempDir
 import java.io.File
+import com.avito.android.build_verdict.RawBuildVerdictWriter.Companion.buildVerdictFileName as rawBuildVerdictFileName
 
 class BuildVerdictPluginTest {
```
Comment (sergioboi) [inline subprojects/gradle/build-verdict/src/test/kotlin/com/avito/android/build_verdict/BuildVerdictPluginTest.kt:16]: I decided to do not to cover the plain text verdict by tests.
Everything may change. I check readability by eyes ))

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/report/Report.kt [hunk 3]
```diff
@@ -231,31 +231,24 @@ interface Report : ReadReport {
         }
 
         override fun sendLostTests(lostTests: List<AndroidTest.Lost>) {
-            logger.critical(
```
Comment (sergioboi) [inline subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/report/Report.kt:234]: unrelated - this causes the build log to contain too many lines at the end
Now you will see this at the build-verdict

File: subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/PlainTextBuildVerdictWriter.kt [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    private val logger: CILogger
+) : BuildVerdictWriter {
+
+    override fun write(buildVerdict: BuildVerdict) {
+        val dir = buildVerdictDir.apply { mkdirs() }
+        val file = File(dir, buildVerdictFileName)
+        file.createNewFile()
+        file.writeText(
+            """
+Your build FAILED: "${buildVerdict.rootError.message}"
```
Comment (sergioboi) [inline subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/PlainTextBuildVerdictWriter.kt:17]: E.g. how the verdict looks

Your build FAILED: "Execution failed for task ':app:kaptDebugKotlin'."
-------------------------------------------------------

Failed tasks:
1: Task :app:kaptDebugKotlin
* What went wrong:
/private/var/folders/tx/0j4xvb9d36xfgv1h3hd4th580000gn/T/junit16968900557237870342/app/build/tmp/kapt3/stubs/debug/DaggerComponent.java:6: error: [Dagger/MissingBinding] CoffeeMaker cannot be provided without an @Inject constructor or an @Provides-annotated method.
public abstract interface DaggerComponent {
                ^
      CoffeeMaker is requested at
          DaggerComponent.maker()
________________________________________

**Review by eugene-krivobokov — APPROVED** [review]

File: subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/PlainTextBuildVerdictWriter.kt [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+                        append(lineSeparator)
+                    }
+                }
+            }""".trimMargin()
+        )
+        logger.warn("Pretty build verdict at $file")
+    }
+
+    companion object {
+        val buildVerdictFileName = "pretty-build-verdict.json"
```
Comment (eugene-krivobokov) [inline subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/PlainTextBuildVerdictWriter.kt:38]: This is not a json. Please, change the extension

**Pull request merged**